services/thermal_mgr/thermal_mgr.c

```diff
@@ -41,20 +41,49 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 }
 
+#define ERR_CODE_FAIL 1
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+  if(xQueueSend(thermalMgrQueueHandle, event, portMAX_DELAY) == pdTRUE){
+    return ERR_CODE_SUCCESS;
+  }
+  return ERR_CODE_FAIL;
 }
-
+#define THERMAL_MGR_EVENT_OS_INTERRUPT 1
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&event);
+
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t *config = (lm75bd_config_t *) pvParameters;
+  uint8_t thermalSensorAddr = config->devAddr;
+  thermal_mgr_event_t event;
+  float temp;
+
+
   while (1) {
-    
+    if(xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdTRUE) {
+      if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD) {
+        if(readTempLM75BD(thermalSensorAddr, &temp) == ERR_CODE_SUCCESS){
+          addTemperatureTelemetry(temp);
+        }
+      }
+      else if (event.type == THERMAL_MGR_EVENT_OS_INTERRUPT){
+        if(readTempLM75BD(thermalSensorAddr, &temp) == ERR_CODE_SUCCESS){
+          if(temp >= 80.0f){
+            overTemperatureDetected();
+          }
+          else if (temp < 75.0f){
```

Comment:
Use the macros defined in the thermal manager header file instead of hardcoding these values